drivers/cleanup.py

```diff
@@ -147,22 +147,35 @@ def runAbortable(func, ret, ns, abortTest, pollInterval, timeOut):
         pid = os.fork()
         if pid:
             startTime = time.time()
-            while True:
-                if resultFlag.test("success"):
-                    Util.log("  Child process completed successfully")
-                    resultFlag.clear("success")
-                    return
-                if resultFlag.test("failure"):
-                    resultFlag.clear("failure")
-                    raise util.SMException("Child process exited with error")
-                if abortTest() or abortSignaled:
-                    os.killpg(pid, signal.SIGKILL)
-                    raise AbortException("Aborting due to signal")
-                if timeOut and time.time() - startTime > timeOut:
-                    os.killpg(pid, signal.SIGKILL)
-                    resultFlag.clearAll()
-                    raise util.SMException("Timed out")
-                time.sleep(pollInterval)
+            try:
+                while True:
+                    if resultFlag.test("success"):
+                        Util.log("  Child process completed successfully")
+                        resultFlag.clear("success")
+                        return
+                    if resultFlag.test("failure"):
+                        resultFlag.clear("failure")
+                        raise util.SMException("Child process exited with error")
+                    if abortTest() or abortSignaled:
+                        os.killpg(pid, signal.SIGKILL)
+                        raise AbortException("Aborting due to signal")
+                    if timeOut and time.time() - startTime > timeOut:
+                        os.killpg(pid, signal.SIGKILL)
```

Comment:
perhaps we could tidyup by having a resultFlag.clearall() in the finally.

Reply:
> perhaps we could tidyup by having a resultFlag.clearall() in the finally.

Do we still care at that point? If we entered the finally: as the result of an exception, the exception will be re-thrown when the finally completes.